pkg/syncer/admissionhandler/cnscsi_admissionhandler.go

```diff
@@ -0,0 +1,99 @@
+package admissionhandler
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crConfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	"sigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service/common"
+	"sigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service/logger"
+)
+
+const (
+	//WebhookTlsMinVersion = "1.2"
+	ValidationWebhookPath            = "/validate"
+	DefaultWebhookPort               = 9883
+	DefaultWebhookMetricsBindAddress = "0"
+)
+
+func GetWebhookPort() int {
+	portStr, ok := os.LookupEnv("CNSCSI_WEBHOOK_SERVICE_CONTAINER_PORT")
+	if !ok {
+		return DefaultWebhookPort
+	}
+
+	result, err := strconv.ParseInt(portStr, 0, 0)
+	if err != nil {
+		panic(fmt.Sprintf("malformed configuration: CNSCSI_WEBHOOK_SERVICE_CONTAINER_PORT, expected int: %v", err))
+	}
+
+	return int(result)
+}
+
+func GetMetricsBindAddress() string {
+	metricsAddr, ok := os.LookupEnv("CNSCSI_WEBHOOK_SERVICE_METRICS_BIND_ADDR")
+	if !ok {
+		return DefaultWebhookMetricsBindAddress
+	}
+
+	return metricsAddr
+}
+
+func startCNSCSIWebhookManager(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+
+	webhookPort := GetWebhookPort()
+	metricsBindAddress := GetMetricsBindAddress()
+	log.Infof("setting up webhook manager with webhookPort %v and metricsBindAddress %v",
+		webhookPort, metricsBindAddress)
+	mgr, err := manager.New(crConfig.GetConfigOrDie(), manager.Options{
+		MetricsBindAddress: metricsBindAddress,
+		Port:               webhookPort})
+	if err != nil {
+		log.Error(err, "unable to set up overall controller manager")
```

Comment:
can we use `log.Fatalf(` to crash container instead of `os.Exit(1)`

Reply:
Yes